File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n=s.size();
        if(n==0) return 0;
        unordered_set<char> seen;
        int left=0,maxi=0;
        for(int right=0;right<n;right++){
            while(seen.count(s[right])){
                seen.erase(s[left]);
                left++;
            }
            seen.insert(s[right]);
            maxi=max(maxi,right-left+1);
        }
        return maxi;

    }
};

File: Linked_List/3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {

        int n = s.length();

        int[] freq = new int[256];

        int maxi=0;

        int left=0,right=0;

        while(right<n){

            freq[s.charAt(right)]++;


            if(freq[s.charAt(right)]>1){

                while(freq[s.charAt(right)]>1){

                    

                    freq[s.charAt(left)]--;
                    left++;

                }
            }

            maxi = Math.max(maxi,right-left+1);

            right++;
        }

        return maxi;
        
    }
}